service/matching/tasklist/task_reader.go

```diff
@@ -211,7 +211,9 @@ getTasksPumpLoop:
 				}
 
 				if len(tasks) == 0 {
-					tr.taskAckManager.SetReadLevel(readLevel)
+					// even though we didn't handle any tasks, we want to advance the ack-level
+					// to avoid needless querying database the next time
+					tr.taskAckManager.SetAckLevel(readLevel)
```

Comment:
when a new task is created at this point, is it guaranteed it is going to have an id > readLevel?

Reply:
It is. That's how task-ID allocation works in taskWriter.
But I also changed the diff for this to be very explicit, and added a test (and the end of it I also check for this for sake of sanity).